docs/source/complex_numbers.rst

```diff
@@ -0,0 +1,155 @@
+.. _complex_numbers-doc:
+
+Complex Numbers
+===============
+
+Complex Numbers are numbers that can be expressed in the form :math:`a + bi`, where a and b are real numbers,
+and *i* is a solution of the equation :math:`x^2 = −1`. Complex numbers frequently occur in mathematics and
+engineering, especially in signal processing. The aim of introducing Complex Tensors is to provide a more natural
+user experience for users and libraries (eg. TorchAudio) that currently workaround the lack of Complex Tensors
+by using Float Tensors with shape :math:`(..., 2)` where the last dimension contains the real and imaginary values.
+
+We support many functions for Complex Tensors eg. SVD, QR etc. Operations on complex tensors are likely to be
+faster than operations on float tensors mimicking them. We also have a `Vec256 class` for `ComplexFloat` and
```

Comment:
Vec256 is a bit internal and I probably wouldn't mention it in a user doc like this

Reply:
Also `ComplexFloat` and `ComplexDouble` haven't been defined yet, but they also sound internal.

You can just say something like "Operations involving complex numbers in PyTorch are optimized to use vectorized assembly instructions and specialized kernels (e.g. LAPACK, CuBlas)."

I'm not really sure why you want to compare users writing their own functions -- why would they do that?